ts/nni_manager/training_service/local/localTrainingService.ts

```diff
@@ -253,7 +253,16 @@ class LocalTrainingService implements TrainingService {
 
             return Promise.resolve();
         }
-        tkill(trialJob.pid, 'SIGKILL');
+        tkill(trialJob.pid, 'SIGTERM');
+        const pid = trialJob.pid;
+        setTimeout(((pid: number): void => {
+            tkill(pid, 'SIGKILL', (err) => {
+                if (err){
+                    this.log.warning(`cancel trial job {pid: ${pid}} failed: ${err.message}`);
```

Comment:
I suddenly thought of a problem.
If the pid get reused between `SIGTERM` and `SIGKILL`, this will kill a random process.
If you don't want to elegantly solve the problem in this release, I suggest to reduce the delay to minimize risk.

Reply:
Yes, and other place use `kill pid` also has this risk, maybe elegant way is let them kill themselves, need discuss. This release reduce the delay to 1 second.